### DISCUSSION TITLE
Autocomplete/intellisense doesn't work for first generic argument until second is provided

### DISCUSSION THREAD

**Discussion by Naught0** [microsoft/pyright#7417] [Q&A] [answered]
I've encountered a strange behavior where the first argument of a function with two generic parameters does not provide autocompletion/intellisense until you begin typing the second parameter.

After typing the second param, I do get intellisense finally in the first.

I've replicated this in vscode, neovim w/ pyright, and online at pyright-play. Can anyone give me insight as to why this is? Am I doing something wrong or is this a bug? 

Code sample in [pyright playground](https://pyright-play.net/?code=GYJw9gtgBALgngBwJYDsDmUkQWEMoAySMApiAIYA2ANFAOIkplIDGtAKoiQGrkgBQ7KAF4onBDz4AKAETsZASn4BpEWK68Qs5Yv4tK5AM6GoAMTBgpDJiFYBtdrWUBdBQC5%2BUL1AAmJYFAARtKGJJTAtDBuYrQA1tHK7p7eKQhGhvz8wBbR5mB2RKQUlHYyaRQQAIwyzrSFZFSl5eQQAEw1zmp5UkpZFgB0wSBAA)

```python
from typing import Literal, Generic, TypeVar
T = TypeVar("T")
K = TypeVar("K")
class Foo(Generic[T, K]):
    def bar(self, t: T, k: K):
        pass

foo: Foo[Literal["param1"], Literal["param2"]] = Foo()

foo.bar("") # completion doesn't work
foo.bar("", "") # completion starts working
```

**Comment by erictraut** [comment, marked as answer]
This appears to be a bug — most likely in the "signature help provider" component. The pylance team is responsible for this code (even though it's in the pyright repo), so I recommend filing a bug in the [pylance-release](https://github.com/microsoft/pylance-release) project.
